src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs

```diff
@@ -4158,26 +4158,34 @@ struct S<out U> { }
     delegate void D<in U>();
     class B<out U> { }
 }";
-
-            ParseAndValidate(test,
-                // (3,12): error CS1960: Invalid variance modifier. Only interface and delegate type parameters can be specified as variant.
-                Diagnostic(ErrorCode.ERR_IllegalVarianceSyntax, "in").WithLocation(3, 12),
-                // (4,12): error CS7002: Unexpected use of a generic name
-                Diagnostic(ErrorCode.ERR_UnexpectedGenericName, "this").WithLocation(4, 12),
-                // (4,17): error CS1960: Invalid variance modifier. Only interface and delegate type parameters can be specified as variant.
-                Diagnostic(ErrorCode.ERR_IllegalVarianceSyntax, "out").WithLocation(4, 17),
```

Comment:
This error should go away.  At parse time we already report that type parameters are not allowed on an indexer.  We don't do any sort of 'variance' check because that's a more specific error that isn't necessary since indexers aren't allowed at all.  And, as the type parameters are added as skipped tokens to the indexer, we don't even really have any type parameter nodes to check later on.  THis new behavior is better.